### PR TITLE
Fixes STIX file loading and lightcurve/spectrogram plotting

### PR DESCRIPTION
**Pull request by natsat0919**
This PR fixes the functions for handling STIX spectral and SRM files, this means that the Sunxspex class will be apple to correctly load in the new/final version of STIX files. I'm mainly fixing formatting of keywords and data, change of time units, format of energy bins and counts that are given in the spectral files. I have also fixed the incorrect plotting of background and event time-ranges for STIX lightcurves and spectrograms (examples in issue #97).

### CODE REVIEW

**Comment by DanRyanIrish** [comment]
Hi @natsat0919.  Thanks so much for contributing this PR.  Your changes appear to be breaking a lot of the tests.  Can you clarify why this is?  Is it possible you've made changes in certain places that a valid for STIX but not other instruments?

**Comment by natsat0919** [comment]
Hi @DanRyanIrish, I'm missing a changelog file that I'll add. I'm still trying to figure out why the other tests are failing, but I think that it's because of changing the fundamental loading function for STIX files and the tests aren't adjusted to that? The changes that I have done were only for STIX (mainly in the stix_spec_code.py), so it shouldn't affect any other instruments

**Comment by settwi** [comment]
Following up on our discussion @natsat0919 and @KriSun95: I think we should be careful and explicit as to which kind of "spectrogram" we load in, i.e. the one produced by IDL or the one from the data center, and pay particular attention to the statistical errors that are being used. I would be happy to review some of the code in this PR and see how we can make sure that distinction is clear. Or we can merge the PR and do that later. Thoughts?

**Comment by natsat0919** [comment]
@settwi I agree, we definitely need to get that sorted as a priority. I think right now is the best to get this out as the basic working version (just updating what we already have) and build up from there? Also the STIX srm and spectral files tests are outdated (hence I got so many fails) so I'm sorting that out as well. Meanwhile, we can think what's the best way to implement the "distinction" when loading in?

**Comment by KriSun95** [comment]
This looks great, thanks @natsat0919 ! I've had a play about with the changes and it works fine for me. Playing about with it has pointed out a small but important bug elsewhere (that I need to fixed) but I can't see anything wrong with this PR. I definitely agree with both you and @settwi about moving on to think about how to separate loading in of the two STIX file formats for the spectra (data centre and IDL).

I'll return to this later today, add a review for the PR, and then move on to approve the merge if there are no other objections.

**Review by KriSun95 — APPROVED** [review]
This looks great and gives us a good foundation to actually start working with/support STIX data. We can use this contribution to test any future changes using real data and use it as a good starting point.

**Comment by DanRyanIrish** [comment]
Do we consider this PR ready to merge?

**Comment by natsat0919** [comment]
@DanRyanIrish yes, I just did some very minor refactoring changes after Kris's approval. It should be ready to merge

File: sunxspex/sunxspex_fitting/stix_spec_code.py
Comment (samaloney) [inline sunxspex/sunxspex_fitting/stix_spec_code.py:44]: I'm not sure this change is needed if the goal is to get the value in seconds `to_value(u.s)` could be used and would remove the somewhat arbitrary `* (10**(-2))` below.

**Review by samaloney — CHANGES_REQUESTED** [review]

File: sunxspex/sunxspex_fitting/stix_spec_code.py
Comment (KriSun95) [reply sunxspex/sunxspex_fitting/stix_spec_code.py:44]: Good catch! The `_ds_times2s_times`  function could be removed altogether, this seems like a relic from when I was coding for multiple file structures. Would it be better to just convert `times_mids` and `time_deltas` (lines 31 and 32, respectively) on the same lines and remove the `_ds_times2s_times` function? Say change line 31 to something like `time_mids = (sdict["2"][1]["time"]<<u.ds).to_value(u.s)` and change line 32 in a similar way?

Comment (natsat0919) [reply sunxspex/sunxspex_fitting/stix_spec_code.py:44]: Oh yeah, that's a good point. Thanks for raising that change. I have just corrected that (the time is in units of cs) by multiplying tds by u.cs and keeping time_mids and time_deltas without units (assigning units to the arrays breaks the code if you try to select an event/background time region).

**Review by samaloney — APPROVED** [review]

**Pull request merged**
